pkg/cloud/openstack/clients/machineservice.go

```diff
@@ -95,6 +100,35 @@ func GetCloudFromSecret(kubeClient kubernetes.Interface, namespace string, secre
 	return clouds.Clouds[cloudName], nil
 }
 
+func authenticatedClient(ao *gophercloud.AuthOptions) (*gophercloud.ProviderClient, error) {
+
+        client, err := openstack.NewClient(ao.IdentityEndpoint)
+        if err != nil {
+                return nil, err
+        }
+        // add HTTPClient includes transport
+	caFile := "/etc//certs/cacert"
+	caCert, err := ioutil.ReadFile(caFile)
+	if err != nil {
+		panic(err)
+	}
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM(caCert)
+
+	tlsConfig := &tls.Config{
+		RootCAs: caCertPool,
+	}
+	tlsConfig.BuildNameToCertificate()
+	transport := &http.Transport{TLSClientConfig: tlsConfig}
+	client.HTTPClient = http.Client{Transport: transport}
+
+	err = openstack.Authenticate(client, *ao)
```

Comment:
forgive me if I misunderstand something here.. I believe https need ca and http doesn't need it
from context we are requesting that to be mandatory https ,maybe I missed something?

Reply:
Yes, you are right. I should have update code https is not mandatory. I will fix.

Reply:
Fixed in the following commit: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/commit/192e70cae1d886113b023b7b763de77e6d8c330a